src/main/java/org/mule/extension/sftp/internal/source/SftpDirectoryListener.java

```diff
@@ -284,29 +284,35 @@ public void onRejectedItem(Result<InputStream, SftpFileAttributes> result, Sourc
   }
 
   private void postAction(PostActionGroup postAction, SourceCallbackContext ctx) {
-    try {
-      postAction.validateSelf();
-    } catch (IllegalArgumentException e) {
-      LOGGER.error(e.getMessage());
+    if (LOGGER.isTraceEnabled()) {
+      try {
+        postAction.validateSelf();
+      } catch (IllegalArgumentException e) {
+        LOGGER.trace(e.getMessage());
+      }
     }
 
     SftpFileSystem fileSystem = ctx.getConnection();
     fileSystem.changeToBaseDir();
     ctx.<SftpFileAttributes>getVariable(ATTRIBUTES_CONTEXT_VAR).ifPresent(attrs -> {
-      try {
-        if (postAction.isAutoDelete()) {
-          fileSystem.delete(attrs.getPath());
-        } else if (postAction.getMoveToDirectory() != null) {
+      if (postAction.getMoveToDirectory() != null) {
+        try {
           fileSystem.move(config, attrs.getPath(), postAction.getMoveToDirectory(), false, true,
                           postAction.getRenameTo());
+        } catch (FileAlreadyExistsException e) {
+          if (postAction.isAutoDelete()) {
+            fileSystem.delete(attrs.getPath());
```

Comment:
don't do this in two places, change the logic so there's only one place where the file can be deleted